sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/windowing/Trigger.java

```diff
@@ -87,7 +88,7 @@ protected Trigger() {
   }
 
   public List<Trigger> subTriggers() {
-    return subTriggers;
+    return MoreObjects.firstNonNull(subTriggers, Collections.emptyList());
```

Comment:
I think also we could enable nullness checking on the subclasses and discover that the field should never be null. Anyhow this fix makes the API null-safe so looks good even though I think it can go further.

Reply:
I agree that it is questionable why the field should ever be null. On the other hand, the code seems to distinguish cases when the field is null and when the list is empty. This might be fixed later with other nullness related fixes.